integration_tests/test_unsigned_02.py

```diff
@@ -0,0 +1,10 @@
+from lpython import u16, i32
+
+# test issue 2170
+
+i : i32
+u : u16 = u16(32768)
+x : i32
+
+for i in range(i32(u)):
+    x = i * 2
```

Comment:
If in case we update the changes in this PR, I suggest we add an assert in this test case (and also possibly a print statement before the assert).